x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/upgrade_job_snapshot.yml

```diff
@@ -17,17 +17,96 @@ setup:
             }
           }
 
+  # It's too hard to create a genuine model snapshot in a YAML test.
+  # All we can do is create the descriptor doc that will allow the
+  # endpoints to get past an existence check. Then the actual snapshot
+  # upgrade will of course fail, but we can test that the stats report
+  # the attempt.
+  - do:
+      headers:
+        Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
+        Content-Type: application/json
+      index:
+        index: .ml-anomalies-shared
+        id:    upgrade-model-snapshot_model_snapshot_1234567890
+        body:  >
+          {
+            "job_id": "upgrade-model-snapshot",
+            "min_version": "7.15.1",
+            "timestamp": "2021-12-13T03:04:05Z",
+            "description": "just for this test",
+            "snapshot_id": "1234567890",
+            "snapshot_doc_count": 1,
+            "latest_record_timestamp": "2021-12-13T02:03:04Z",
+            "latest_result_timestamp": "2021-12-13T01:02:03Z",
+            "retain":true
+          }
+
+  - do:
+      headers:
+        Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
+        Content-Type: application/json
+      index:
+        index: .ml-state-000001
+        id:    "upgrade-model-snapshot_model_state_1234567890#1"
+        body:  >
+          {
+          }
+
+  - do:
+      headers:
+        Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
+      indices.refresh:
+        index: [.ml-anomalies-shared,.ml-state-000001]
+
 ---
 "Test with unknown job id":
   - do:
       catch: missing
       ml.upgrade_job_snapshot:
         job_id: "non-existent-job"
         snapshot_id: "san"
+
 ---
 "Test with unknown snapshot id":
   - do:
       catch: missing
       ml.upgrade_job_snapshot:
         job_id: "upgrade-model-snapshot"
-        snapshot_id: "snapshot-9999"
+        snapshot_id: "9999999999"
+
+---
+"Test stats all snapshots":
```

Comment:
Currently there are no requests to the `get_model_snapshot_upgrade_stats` API in YAML tests that aren't either a 404 or skipped. Could we add a test that isn't skipped that has a valid response so the clients team can create specs for and test the response structure?

Reply:
Adding that test in this PR would involve fixing #81578 in this same PR.  I was going to fix that and unmute the muted test at the end of the file in a separate followup PR this week, just to keep the sizes of the PRs manageable.

The YAML test framework isn't rich enough to test upgrade of a model snapshot that's created by the test itself.  But the test at the end that gets the stats for upgrade of the corrupt model snapshot will return the same fields that getting stats for upgrade of a valid snapshot would get.  So could you wait until the end of the week before creating specs and testing the response structure?

Reply:
Merging this and fixing the test later works for me! I figured that was the plan.

Reply:
@sethmlarson the test that should give the flight recorder a response it can get the structure from is unmuted in #81831 - see https://github.com/elastic/elasticsearch/pull/81831/files#diff-0a85b50d1fe60d5fcfbfabef2bb1a3cc7047b5f86374e8279f82a343ee4e6f03L96-L98.